src/components/Chat/index.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useRef, useState } from "react";
+
+import firebase from "firebase/app";
+import "firebase/auth";
+import "firebase/analytics";
+import "./index.css";
+
+import { useAuthState } from "react-firebase-hooks/auth";
+import { useCollectionData } from "react-firebase-hooks/firestore";
+
+const auth = firebase.auth();
+const firestore = firebase.firestore();
+
+function Chat() {
+  const [user] = useAuthState(auth);
+
+  return (
+    <div>
+      <header>
+        <SignOut />
+      </header>
+
+      <section>{user ? <ChatRoom /> : <SignIn />}</section>
+    </div>
+  );
+}
+
+function SignIn() {
```

Comment:
I think you can import and use the sign in and sign out created by orcun before instead of doing them again

Reply:
The current sign in doesn't work properly, that's why i need to add this. After this is merged, will try to take this outside and use everywhere